packages/protocol/test/ACE/validators/burn/index.js

```diff
@@ -0,0 +1,252 @@
+        });
+
+        it('validate that burned notes of zero value work', async () => {
+            const noteValues = [50, 30, 0, 20];
+            const numNotes = noteValues.length;
+            const aztecAccounts = [...new Array(numNotes)].map(() => secp256k1.generateAccount());
+            const notes = aztecAccounts.map(({ publicKey }, i) => {
+                return note.create(publicKey, noteValues[i]);
+            });
+
+            const newTotalBurned = notes[0];
+            const oldTotalBurned = notes[1];
+            const adjustedNotes = notes.slice(2, 4);
+
+            const senderAddress = accounts[0];
+
+            const { proofData, expectedOutput } = burn.encodeBurnTransaction({
+                newTotalBurned,
+                oldTotalBurned,
+                adjustedNotes,
+                senderAddress: accounts[0],
+            });
+
+            const opts = {
+                from: senderAddress,
+                gas: 4000000,
+            };
+            const result = await adjustSupplyContract.validateAdjustSupply(
+                proofData,
+                senderAddress,
+                constants.CRS,
+                opts
+            );
+
+            expect(result).to.equal(expectedOutput);
+
+            const gasUsed = await adjustSupplyContract.validateAdjustSupply.estimateGas(
+                proofData,
+                senderAddress,
+                constants.CRS,
+                opts
+            );
+
+            console.log('gas used = ', gasUsed);
+        });
+    });
+
+    describe('failure states', () => {
+        beforeEach(async () => {
+            adjustSupplyContract = await AdjustSupply.new({
+                from: accounts[0],
+            });
+        });
+
+        it('validates failure for unbalanced input/output notes', async () => {
+            const noteValues = [50, 30, 40, 10];
+            const numNotes = noteValues.length;
+            const aztecAccounts = [...new Array(numNotes)].map(() => secp256k1.generateAccount());
+            const notes = aztecAccounts.map(({ publicKey }, i) => {
+                return note.create(publicKey, noteValues[i]);
+            });
+
+            const newTotalBurned = notes[0];
+            const oldTotalBurned = notes[1];
+            const adjustedNotes = notes.slice(2, 4);
+            const senderAddress = accounts[0];
+
+            const { proofData } = burn.encodeBurnTransaction({
+                newTotalBurned,
+                oldTotalBurned,
+                adjustedNotes,
+                senderAddress: accounts[0],
+            });
+
+            const opts = {
+                from: accounts[0],
+                gas: 4000000,
+            };
+
+            await truffleAssert.reverts(adjustSupplyContract.validateAdjustSupply(proofData, senderAddress, constants.CRS, opts));
+        });
+
+        it('validates failure when using a fake challenge and fake proof data', async () => {
```

Comment:
We should test that this throws if there are 0 output notes.

Reply:
Moved to testing checklist